packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -222,14 +224,74 @@ const duplicateNameError = async (context: HookContext<LocationService>) => {
   }
 }
 
+/* BEFORE FIND HOOKS */
+/**
+ * @function rejectEmptyQuery
+ * @param context
+ * @description This hook function ensure that no empty queries are present
+ */
+const rejectEmptyQuery = async (context: HookContext<LocationService>) => {
+  const keys: string[] = Object.keys(context.params.query || {})
+  if (keys.length == 0) throw new BadRequest(`Query can't have empty params object`)
+}
+/**
+ * @function restrictProtectedQuery
+ * @param context
+ * @description This hook function restricts allowed query params for scoped user's calls
+ * from authorized sections (admin, dashboard, studio)
+ */
+const restrictProtectedQuery = async (context: HookContext<LocationService>) => {
+  keepQuery(
+    'id',
+    'name',
+    'slugifiedName',
+    '$like',
+    '$limit',
+    '$sort',
+    '$select',
+    '$skip',
+    'sceneId',
+    'projectId'
+  )(context)
+}
+/**
+ * @function restrictPublicQuery
+ * @param context
+ * @description This hook function restricts allowed query params for public calls
+ * from the viewer section by only allowing the slugifiedName parameter, any other
+ * parameter will be discarded from the query protecting from any malicious query injection.
+ *
+ * This function is discarding all params but slugifiedName instead of throwing a BadRequest if other
+ * parameters are present, having as goal to avoid providing any feedback to malicious users
+ * that could allow them to figure out unforeseen and unsafe patterns or combinations in order to
+ * successfully perform an injection attack.
+ */
+const restrictPublicQuery = async (context: HookContext<LocationService>) => {
+  keepQuery('slugifiedName')(context)
+}
+
 export default {
   around: {
     all: [schemaHooks.resolveExternal(locationExternalResolver), schemaHooks.resolveResult(locationResolver)]
   },
 
   before: {
     all: [schemaHooks.validateQuery(locationQueryValidator), schemaHooks.resolveQuery(locationQueryResolver)],
-    find: [discardQuery('action'), discardQuery('studio'), sortByLocationSetting],
+    find: [
+      // Empty query object in findLocation is not allowed
+      rejectEmptyQuery,
+      // Persist query to avoid hook collitions or malformed knex query instances
+      persistQuery,
+      // Clean up query of non db relevant parameters
+      discardQuery('action'),
+      discardQuery('studio'),
+      // On external calls. evaluate action property presence and restrict queries according
+      iff(
+        isProvider('external'),
+        iffElse(hasAction, [verifyScope('location', 'read'), restrictProtectedQuery], [restrictPublicQuery])
```

Comment:
So it seems like hasAction is running `verifyScope('location', 'read')` for all actions which is wrong.

`verifyScope('location', 'read')` should work only if user is an admin. If a user has `location:read` then he can read all locations in our system. Even those that belong to another user.

For end users using studio and dashboard, they only have `editor:write` scope. So `verifyScope('location', 'read')` should only be performed for `action: admin` requests and not other actions.

Right now, if you were testing this flow with account created through seed data locally, then that user has all admin scopes as well as a MT account. For testing this scenarios. Try to open an incognito window and do a new user signup. Then try accessing locations from studio and dashboard. Because this user will only have `editor:write` scope and truly represents an end user.

Reply:
@hanzlamateen I've always used the sign up section to create users, don't know how to create seeded users, regardless I cleaned up my environment `npm run dev-reinit`, I ran the project, using magic link I created a new user from Sign Up:

![image](https://github.com/user-attachments/assets/de7eadba-7aa4-468c-aab6-54212d751f5e)
(See ID: 43b3fd8c-3f89-4c39-870b-6909d62df8af)

Then if I go and see the scopes associated with this user

![image](https://github.com/user-attachments/assets/7eb24e49-ff63-4cf6-a3ce-d12f5ab94dda)

Am I missing something? I just want to understand because you said this user will only have editor:write.

Aside of that I did test the different locations related sections (admin/location, dashboard/location, studio/publish location "viewer") and everything was working as expected before I sending the PR last night.

Anyway I still made the change you requested in case that I'm missing something

```ts
iff(isAction('admin'), verifyScope('location', 'read')).else(verifyScope('editor', 'write'))
```
Which after my testes and as I assumed it would work exactly the same because of this:

![image](https://github.com/user-attachments/assets/359a306f-52b6-4365-b556-c50637ac7fb8)

In deed it worked just the same before the changes, so I don't mind pushing the changes, I just want to understand the difference because I don't see any since both scopes `location:read` and `editor:write` are associated with the user I just created and not just `editor:write` as you said it should be.

Reply:
So, after dev-reinit. The first user is automatically made an admin. I hope you have your .env.local setup as mentioned in readme below.

https://github.com/theinfinitereality/irpro-multitenancy/blob/dev/readme.md
![image](https://github.com/user-attachments/assets/79038f8d-535c-428c-a492-d4d99d56eacc)

The screenshot you showed had 1st user getting all the scope, which does not depict a typical end user. So once you have this user.

Open an incognito window and there signup using a different SSO/Email. Now this user will be 2nd user in the system and it should only have editor:write scope and it depicts a real end user using our platform.